docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -22,110 +24,58 @@ command:
-
-     .. code-block:: text
-
-         node.internal-address-source=FQDN
+.. code-block:: text
 
-   - It is also possible to specify each node's fully-qualified hostname manually.
-     This is different for every host. Hosts should be in the same domain to
-     make it easy to create the correct SSL/TLS certificates.
-     e.g.: ``coordinator.example.com``, ``worker1.example.com``, ``worker2.example.com``.
+    http-server.https.enabled=true
+    http-server.https.port=<https port>
 
-     .. code-block:: text
+Now you can enable TLS for internal communication with the following
+configuration identical on all cluster nodes.
 
-         node.internal-address=<node fqdn>
+1. Configure a shared secret for internal communication as described in
+   the preceding section.
 
+2. Enable automatic certificate creation and trust setup in
+   ``etc/config.properties``:
 
-3. Generate a Java Keystore File. Every Trino node must be able to connect to
-   any other node within the same cluster. It is possible to create unique
-   certificates for every node using the fully-qualified hostname of each host,
-   create a keystore that contains all the public keys for all of the hosts,
-   and specify it for the client (see step #8 below). In most cases it is
-   simpler to use a wildcard in the certificate as shown below.
+   .. code-block:: properties
 
-   .. code-block:: text
+     internal-communication.https.required=true
 
-       keytool -genkeypair -alias example.com -keyalg RSA -keystore keystore.jks
-       Enter keystore password:
-       Re-enter new password:
-       What is your first and last name?
-         [Unknown]:  *.example.com
-       What is the name of your organizational unit?
-         [Unknown]:
-       What is the name of your organization?
-         [Unknown]:
-       What is the name of your City or Locality?
-         [Unknown]:
-       What is the name of your State or Province?
-         [Unknown]:
-       What is the two-letter country code for this unit?
-         [Unknown]:
-       Is CN=*.example.com, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
-         [no]:  yes
-
-   .. Note: Replace `example.com` with the appropriate domain.
-
-4. Distribute the Java Keystore File across the Trino cluster.
-
-5. Enable the HTTPS endpoint.
+   If you upgrade from the manual setup of an older version, you need to ensure
+   that ``internal-communication.https.keystore.path`` and
+   ``internal-communication.https.truststore.path`` are not set.
 
-   .. code-block:: text
+3. Change the URI for the discovery service to use HTTPS in
+   ``etc/config.properties``:
 
-       http-server.https.enabled=true
-       http-server.https.port=<https port>
-       http-server.https.keystore.path=<keystore path>
-       http-server.https.keystore.key=<keystore password>
+   .. code-block:: properties
 
-6. Change the discovery uri to HTTPS.
-
-   .. code-block:: text
+     discovery.uri=https://<coordinator fqdn>:<https port>
```

Comment:
This currently must be an IP address. Hostname will not work in the current code.

Reply:
I confirmed with @dain that it's intentional to only support IP address, in order to not conflict with an explicitly configured certificate used for the coordinator.

Reply:
Okay.. updated text.